scripts/nbconverted/high_weight_genes.py

```diff
@@ -0,0 +1,355 @@
+                                   color=algorithm_color_dict['VAE_300']);
+
+
+# ## Observe distribution of skewness and kurtosis
+# 
+# * Skewness - measure of distribution symmetry (normal distribution = 0)
+# * Kurtosis - measure of tail density (light tail is < 0, heavy tail is > 0)
+
+# In[21]:
+
+pca_node_categories = pca_weights.get_node_categories(melted=True)
+ax, pca_outlier = pca_weights.plot_skewkurtosis(color=algorithm_color_dict['PCA'])
+print(pca_outlier)
+
+
+# In[22]:
+
+ica_node_categories = ica_weights.get_node_categories(melted=True)
+ax, ica_outlier = ica_weights.plot_skewkurtosis(color=algorithm_color_dict['ICA'])
+print(ica_outlier)
+
+
+# In[23]:
+
+nmf_node_categories = nmf_weights.get_node_categories(melted=True)
+ax, nmf_outlier = nmf_weights.plot_skewkurtosis(color=algorithm_color_dict['NMF'])
+print(nmf_outlier)
+
+
+# In[24]:
+
+adage_node_categories = adage_weights.get_node_categories(melted=True)
+ax, adage_outlier = adage_weights.plot_skewkurtosis(color=algorithm_color_dict['ADAGE'])
+print(adage_outlier)
+
+
+# In[25]:
+
+tybalt_node_categories = tybalt_weights.get_node_categories(melted=True)
+ax, tybalt_outlier = tybalt_weights.plot_skewkurtosis(color=algorithm_color_dict['Tybalt'])
+print(tybalt_outlier)
+
+
+# In[26]:
+
+vae_2l100_node_categories = vae_2l100_weights.get_node_categories(melted=True)
+ax, vae_2l100_outlier = vae_2l100_weights.plot_skewkurtosis(color=algorithm_color_dict['VAE_100'])
+print(vae_2l100_outlier)
+
+
+# In[27]:
+
+vae_2l300_node_categories = vae_2l300_weights.get_node_categories(melted=True)
+ax, vae_2l300_outlier = vae_2l300_weights.plot_skewkurtosis(color=algorithm_color_dict['VAE_300'])
+print(vae_2l300_outlier)
+
+
+# ## Define Node Categories based on Feature Activation Distributions
+# 
+# |      |        | Skew   | |
+# | :--: | :----: | :----: | :--: |
+# | **Kurtosis** | | _Low_    | _High_ |
+# |      | _Low_ | Type A | Type C |
+# |      | _High_ | Type B | Type D |
+# 
+# We also define subsets of Type C and Type D nodes because the skew could be positive (left tail) or negative (right tail).
+# 
+# High Skewness and Kurtosis determined statistically significantly different than normal using `scipy.stats.skewtest` and `scipy.stats.kurtosistest` with a Bonferonni adjusted p value.
+# 
+# ### Visualize the distribution of each of these node types across agorithms.
+
+# In[28]:
+
+# Concatenate and visualize node categories
+node_categories = pd.concat([
+        pca_node_categories,
+        ica_node_categories,
+        nmf_node_categories,
+        adage_node_categories,
+        tybalt_node_categories,
+        vae_2l100_node_categories,
+        vae_2l300_node_categories])
+node_categories.head(2)
+
+
+# In[29]:
+
+g = sns.barplot('node_type', y="count", hue='algorithm', palette=algorithm_color_dict,
+                ci=None, data=node_categories)
+
+
+# ### Adjust for node type distributions and reassign high weights
```

Comment:
I would remind the reader about how you make these adjustments at this point. I also find myself wondering what these post-adjustment plots would _ideally_ look like. May be hard to say upstream of any kind of biological (pathway) evaluation, but probably worth thinking about. I would also keep an eye out for how the number of high weight genes output might play into the pathway analyses -- I can see someone making an argument for just picking the top 50 genes and calling it a day, for example (not that I am that someone).

Reply:
> I would remind the reader about how you make these adjustments at this point.

:+1: 

>  I also find myself wondering what these post-adjustment plots would ideally look like. May be hard to say upstream of any kind of biological (pathway) evaluation, but probably worth thinking about.

That is an interesting thought, but i do not think they should look like anything - especially considering the learned features are algorithm dependent and i wouldn't think biology _should_ look like neat and clean distributions. I think upstream analyses with pathway coverage should address this. 

Could this kind of dynamic gene selection within features be a moot point if we decide to use a weighted pathway method (like GSEA)?. Again, a pathway coverage upstream analysis would be telling.

I am also thinking about some sort of simulation analysis where I emulate the node type distribution artificially with gold standard positive and negative gene weights. Thinking about this study design for part of brain trust discussion.

> I can see someone making an argument for just picking the top 50 genes and calling it a day, for example (not that I am that someone).

This is a good reminder of a more "standard" (and probably lazy) analysis and should be considered in any upstream pathway coverage analysis

Reply:
> Could this kind of dynamic gene selection within features be a moot point if we decide to use a weighted pathway method (like GSEA)?. Again, a pathway coverage upstream analysis would be telling.

Yes, this is the kind of thing I was thinking about.

> This is a good reminder of a more "standard" (and probably lazy) analysis and should be considered in any upstream pathway coverage analysis (see #97).

It is certainly less principled.

> I am also thinking about some sort of simulation analysis where I emulate the node type distribution artificially with gold standard positive and negative gene weights. Thinking about this study design for part of brain trust discussion.

Looking forward to this discussion.